src/csharp/Grpc.Core.Api/Status.cs

```diff
@@ -31,48 +31,59 @@ public struct Status
         /// </summary>
         public static readonly Status DefaultCancelled = new Status(StatusCode.Cancelled, "");
 
-        readonly StatusCode statusCode;
-        readonly string detail;
-
         /// <summary>
         /// Creates a new instance of <c>Status</c>.
         /// </summary>
         /// <param name="statusCode">Status code.</param>
         /// <param name="detail">Detail.</param>
-        public Status(StatusCode statusCode, string detail)
+        public Status(StatusCode statusCode, string detail) : this(statusCode, detail, null)
         {
-            this.statusCode = statusCode;
-            this.detail = detail;
         }
 
         /// <summary>
-        /// Gets the gRPC status code. OK indicates success, all other values indicate an error.
+        /// Creates a new instance of <c>Status</c>.
         /// </summary>
-        public StatusCode StatusCode
+        /// <param name="statusCode">Status code.</param>
+        /// <param name="detail">Detail.</param>
+        /// <param name="debugErrorString">Optional internal error string.</param>
+        public Status(StatusCode statusCode, string detail, string debugErrorString)
         {
-            get
-            {
-                return statusCode;
-            }
+            StatusCode = statusCode;
+            Detail = detail;
+            DebugErrorString = debugErrorString;
         }
 
+        /// <summary>
+        /// Gets the gRPC status code. OK indicates success, all other values indicate an error.
+        /// </summary>
+        public StatusCode StatusCode { get; }
+
         /// <summary>
         /// Gets the detail.
         /// </summary>
-        public string Detail
-        {
-            get
-            {
-                return detail;
-            }
-        }
+        public string Detail { get; }
+
+        /// <summary>
+        /// In case of an error, this field may contain additional error details to help with debugging.
+        /// This field will be only populated on a client and its value is generated locally,
+        /// based on the internal state of the gRPC client stack (i.e. the value is never sent over the wire).
```

Comment:
optional: consider a test case to show how it's not transmitted from server to client

Reply:
added.